### PR TITLE
Fix leaks

### PR DESCRIPTION
**Pull request by qnikst**
Commits fixes possible memory leaks in the project.

### CODE REVIEW

**Comment by qnikst** [comment]
@facundominguez please check if memory leaks is fixed in your example. Those were able to fix all problems that I had in my test.

**Comment by mboes** [comment]
@facundominguez confirm?

**Comment by facundominguez** [comment]
@qnikst, How about we add some tests for this?

It doesn't make much difference in the tests I'm trying.

File: src/Network/Transport/InMemory.hs [hunk 1]
```diff
@@ -108,7 +108,7 @@ createTransportExposeInternals = do
 apiNewEndPoint :: TVar TransportState
                -> IO (Either (TransportError NewEndPointErrorCode) EndPoint)
 apiNewEndPoint state = handle (return . Left) $ atomically $ do
-  chan <- newTChan
+  chan <- newBroadcastTChan
```
Comment (facundominguez) [inline src/Network/Transport/InMemory.hs:111]: How does this help?

**Pull request closed without merging**